codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -54,7 +54,11 @@
  */
 public class DefaultHttp2Connection implements Http2Connection {
 
-    private final Set<Listener> listeners = new HashSet<Listener>(4);
+    /**
+     * We chose a {@link List} over a {@link Set} to avoid allocating an {@link Iterator} objects when iterating over
+     * the listeners.
+     */
+    private final List<Listener> listeners = new ArrayList<Listener>(4);
```

Comment:
I was also looking at this yesterday :)
1 for local flow controller
1 for remote flow controller
1 for Decompressor
1 for Compressor
1 for HttpToHttp2 adapter

I think 4 is reasonable default.  If anything I would consider adding an argument to the constructor in-case the user wants to bump it up...but not sure if we need this.